pandas/core/groupby.py

```diff
@@ -2586,6 +2586,15 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     """
     group_axis = obj._get_axis(axis)
 
+    if key is None and level is not None and \
+       isinstance(group_axis, MultiIndex):
+        if is_list_like(level) and len(level) == 1:
+            level = level[0]
+
+        if is_scalar(level):
+            key = group_axis.get_level_values(level)
```

Comment:
put a comment on what is going on here. *maybe* we can incorporate this below as well.

I am trying to remove as many special cases as possible.